lib/metrics/events.js

```diff
@@ -175,6 +177,7 @@ module.exports = log => {
           log.flowEvent(Object.assign({}, data, {
             event: 'flow.complete'
           }))
+          amplitude('flow.complete', request, optionalData, data)
```

Comment:
This is why the `complete` events show up before their triggers in the log output: because nothing leaks back to the caller about whether `flow.complete` has been emitted. Although actually, now I write that out, I just had an idea. Let me try something quickly...